src/util/find_uid.c

```diff
@@ -325,9 +329,86 @@ static errno_t get_active_uid_linux(hash_table_t *table, uid_t search_uid)
     return ret;
 }
 
-errno_t get_uid_table(TALLOC_CTX *mem_ctx, hash_table_t **table)
+#ifdef __FreeBSD__
+static errno_t get_active_uid_freebsd(hash_table_t *table, uid_t uid)
+{
+    struct kinfo_proc *kp;
+    hash_key_t key;
+    hash_value_t value;
+    size_t sz;
+    int err, mib[3];
+
+    mib[0] = CTL_KERN;
+    mib[1] = KERN_PROC;
+    mib[2] = KERN_PROC_PROC;
+
+    sz = 0;
+    err = sysctl(mib, 3, NULL, &sz, NULL, 0);
+    if (err) {
+        err = errno;
+        DEBUG(SSSDBG_CRIT_FAILURE, "sysctl failed.\n");
+        return err;
+    }
+    sz *= 2;
+
+    kp = talloc_size(NULL, sz);
+    if (kp == NULL) {
+        DEBUG(SSSDBG_CRIT_FAILURE, "talloc failed.\n");
+        return ENOMEM;
+    }
+
+    err = sysctl(mib, 3, kp, &sz, NULL, 0);
+    if (err) {
+        err = errno;
+        DEBUG(SSSDBG_CRIT_FAILURE, "sysctl failed.\n");
+        talloc_free(kp);
+        return err;
+    }
+
+    err = table != NULL ? 0 : ENOENT;
```

Comment:
Please wrap in `( )` and use `EOK` instead of `0` (just for the sake of consistency).